### ISSUE TITLE
map fitBounds

### ISSUE THREAD

**Issue by sroettering** [Wykks/ngx-mapbox-gl#74]
I can't get fitBounds to work as I expected it to work.
Look at this stackblitz example: https://stackblitz.com/edit/angular-1szcst
Setting fitBounds to a static value does not trigger anything.
Also using a Subject with multiple values over time does not seem to work.
Only if I introduce a button which triggers a new fitBounds value the map starts moving.

Am I doing something wrong? Is fitBounds intended to work like this?

**Comment by Crylion** [comment]
I'm just now trying to use fitBounds and I'm having the same problem :/
I also tried calling `map.fitBounds()` but still nothing happens.
I didn't try testing it with a button, but with a timeout that triggers the fitBounds call (or variable assignment) way after my markers and layers are rendered. But nothing worked :/ 

@sroettering I tried exchanging the BehaviourSubject line in your example with `this.map.fitBounds([9.1, 48.1, 10.1, 49.1]);` AND `this.bounds = [9.1, 48.1, 10.1, 49.1]` in the button press event and both worked as well. So at least with a button, all ways of setting the bounds seem to work.

Could it have anything to to with the rendering of new geoJSON objects and some kind of timing conflicts? In my own application I try to set the bounds after setting some new geoJSON data for the sources/layers to render on the map. Would I have to wait for that to finish somehow?
